.github/workflows/tests.yaml

```diff
@@ -48,14 +48,20 @@ jobs:
     - name: run unit tests
       run: make test.unit
 
+
     - name: Upload coverage to Codecov
-      uses: codecov/codecov-action@v3
+      if: steps.detect_if_should_run.outputs.result == 'true'
+      uses: Wandalen/wretry.action@v1.0.36
       with:
-        token: ${{ secrets.CODECOV_TOKEN }}
-        fail_ci_if_error: true
```

Comment:
Alternatively, we could use `fail_ci_if_error: false` and accept that it may in some cases fail. <controversy> Codecov is not that critical in the end to make it a required check. </controversy> WDYT? I'd prefer that as it's just simpler.

Reply:
I thought about this too but then what's the point of those reports if we don't care about whether they are uploaded or not? Either commit to having them or just straight up abort uploading them.

Reply:
Yeah, I get your point. Let's maybe just add a comment above the step to explain why we want to retry codecov action? Noting that "given it relies on an external service that has proven to temporarily fail, we don't want it to affect the results of CI" would suffice.

Reply:
Added the comment. PTAL